src/ghostfat.c

```diff
@@ -269,23 +275,28 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
     for ( uint32_t i = 0; i < FAT_ENTRIES_PER_SECTOR; ++i )
     {
       // Generate the FAT chain for the firmware "file"
-      uint32_t v = (sectionIdx * FAT_ENTRIES_PER_SECTOR * BPB_SECTORS_PER_CLUSTER) + (i * BPB_SECTORS_PER_CLUSTER);
+      uint32_t v = (sectionRelativeSector * FAT_ENTRIES_PER_SECTOR * BPB_SECTORS_PER_CLUSTER) + (i * BPB_SECTORS_PER_CLUSTER);
 
-      if ( UF2_FIRST_SECTOR <= v && v <= UF2_LAST_SECTOR )
+      if ( UF2_FIRST_CLUSTER_NUMBER <= v && v < UF2_LAST_CLUSTER_NUMBER )
       {
-        ((uint16_t*) (void*) data)[i] = v == UF2_LAST_SECTOR ? 0xffff : (v / BPB_SECTORS_PER_CLUSTER) + 1;
+        ((uint16_t*) (void*) data)[i] = v + 1; // contiguous file, so point to next cluster number
```

Comment:
slip up the last sector make it much easier to read, though the value of `v` doesn't seem to be the same meaning. Should we drop the multiply with `BPB_SECTORS_PER_CLUSTER` since it is cluster num and not sector num

Reply:
<details><summary>Good question.... Let me think this through a bit.</summary><P/>

`sectionRelativeSector` here is the sector offset within the FAT table.
`i` here is the sector-specific FAT entry in the table (treating it as an array for just this single sector).

Therefore, `v` should simply be `sectionRelativeSector * FAT_ENTRIES_PER_SECTOR`, so it reflects the index into the FAT (across all sectors).
</details>

I think the right fix is to modify the line that declares `v` as follows:
```C
uint32_t v = (sectionRelativeSector * FAT_ENTRIES_PER_SECTOR)
```

The rest of the code seems to remain.